internal/active/active.go

```diff
@@ -39,14 +59,37 @@ func CheckIfNodeActive(node models.Node, repos *repositories.Repos) (bool, error
 		metrics.FinalizedBlockHeight <= (latestBlockMetrics.FinalizedBlockHeight - AllowedBlocksBehind) {
 		log.Debugf(
 			"Node %s not active as metrics check failed. " +
-			"Node metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d] " +
-			"Best pool metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d]",
-			node.ID,
+				"Node metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d] " +
+				"Best pool metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d]",
+			nodeID,
 			metrics.BestBlockHeight, metrics.FinalizedBlockHeight,
 			latestBlockMetrics.BestBlockHeight, latestBlockMetrics.FinalizedBlockHeight,
 		)
 		return false, nil
 	}
-	
+
 	return true, nil
 }
+
+// ActivateNodeIfReady sets node to active nodes if latest metrics are valid and node is not penalized
+func ActivateNodeIfReady(nodeID string, repos repositories.Repos) error {
+	node, err := repos.NodeRepo.FindByID(nodeID)
```

Comment:
Why not use IsNodeOnCooldown method here?

Reply:
Because I needed `node` object for `AddNodeToActive`, but now I refactored `AddNodeToActive` and `RemoveNodeFromActive` to have for argument node ID (`string`) and not node struct. I also refactored this here to use `IsNodeOnCooldown`